### ISSUE TITLE
(Check whether your GraphDef-interpreting binary is up to date with your GraphDef-generating binary.)

### ISSUE THREAD

**Issue by dm961027** [tensorflow/tensorflow#32175]
2019-09-02 15:57:27.737 14560-14560/org.tensorflow.demo E/tensorflow: CameraActivity: Exception!
    java.lang.RuntimeException: Failed to load model from 'file:///android_asset/ssd.pb'
        at org.tensorflow.contrib.android.TensorFlowInferenceInterface.<init>(TensorFlowInferenceInterface.java:113)
        at org.tensorflow.demo.TensorFlowObjectDetectionAPIModel.create(TensorFlowObjectDetectionAPIModel.java:91)
        at org.tensorflow.demo.DetectorActivity.onPreviewSizeChosen(DetectorActivity.java:157)
        at org.tensorflow.demo.CameraActivity.onPreviewFrame(CameraActivity.java:120)
        at android.hardware.Camera$EventHandler.handleMessage(Camera.java:1209)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6669)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: java.io.IOException: Not a valid TensorFlow Graph serialization: NodeDef mentions attr 'half_pixel_centers' not in Op<name=ResizeBilinear; signature=images:T, size:int32 -> resized_images:float; attr=T:type,allowed=[DT_INT8, DT_UINT8, DT_INT16, DT_UINT16, DT_INT32, DT_INT64, DT_BFLOAT16, DT_HALF, DT_FLOAT, DT_DOUBLE]; attr=align_corners:bool,default=false>; NodeDef: {{node Preprocessor/map/while/ResizeImage/resize/ResizeBilinear}}. (Check whether your GraphDef-interpreting binary is up to date with your GraphDef-generating binary.).
        at org.tensorflow.contrib.android.TensorFlowInferenceInterface.loadGraph(TensorFlowInferenceInterface.java:561)
        at org.tensorflow.contrib.android.TensorFlowInferenceInterface.<init>(TensorFlowInferenceInterface.java:105)
        at org.tensorflow.demo.TensorFlowObjectDetectionAPIModel.create(TensorFlowObjectDetectionAPIModel.java:91) 
        at org.tensorflow.demo.DetectorActivity.onPreviewSizeChosen(DetectorActivity.java:157) 
        at org.tensorflow.demo.CameraActivity.onPreviewFrame(CameraActivity.java:120) 
        at android.hardware.Camera$EventHandler.handleMessage(Camera.java:1209) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.app.ActivityThread.main(ActivityThread.java:6669) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)

**Comment by dm961027** [comment]
when I use my .pb file to replace detectoractivity.java , I have above question.

my replace:

private static final int TF_OD_API_INPUT_SIZE = 300;
 private static final String TF_OD_API_MODEL_FILE = "file:///android_asset/ssd.pb";
  private static final String TF_OD_API_LABELS_FILE = "file:///android_asset/label_list.txt";

**Comment by oanush** [comment]
@dm961027 ,
Please provide details about what platform you are using (operating system, architecture). Also include your TensorFlow version.

In order to expedite the trouble-shooting process, Can you share a simple and standalone code to reproduce the issue reported here? Thanks!

**Comment by dm961027** [comment]
I use  Windows10  , Android studio 3.3 ，the demo in tensorflow_ android，modify the file "DetectorActivity.java" 
my .pb file use Linux system  train ，use tensorflow 1.14.0
 
android build gradle 

nativeBuildSystem = 'none'

dependencies {
    if (nativeBuildSystem == 'cmake' || nativeBuildSystem == 'none') {
        implementation 'org.tensorflow:tensorflow-android:+'
          }

**Comment by dm961027** [comment]
是不是我训练数据的tensorflow版本太高了，应该使用tensorflow 1.13.0以下的版本

**Comment by sjtusmartboy** [comment]
@dm961027 欺负人家不懂中文吗[奸笑]

**Comment by dm961027** [comment]
help me , please

**Comment by jvishnuvardhan** [comment]
@dm961027 Could you summarize your issue with little more details? Thanks

**Comment by dm961027** [comment]
@jvishnuvardhan  when i build my app,there is no error, but when i run my app ,the error is the one when i refer to , so i do not know how to summarize more details .That is all my error ,sorry 



2019-09-15 11:38:39.830 19087-19087/? I/tensorflow.dem: Not late-enabling -Xcheck:jni (already on)
2019-09-15 11:38:39.884 19087-19087/? W/tensorflow.dem: Unexpected CPU variant for X86 using defaults: x86
2019-09-15 11:38:40.279 19087-19087/org.tensorflow.demo I/tensorflow.dem: The ClassLoaderContext is a special shared library.
2019-09-15 11:38:40.627 19087-19087/org.tensorflow.demo W/tensorflow.dem: JIT profile information will not be recorded: profile file does not exits.
2019-09-15 11:38:40.647 19087-19087/org.tensorflow.demo I/chatty: uid=10102(org.tensorflow.demo) identical 2 lines
2019-09-15 11:38:40.647 19087-19087/org.tensorflow.demo W/tensorflow.dem: JIT profile information will not be recorded: profile file does not exits.
2019-09-15 11:38:40.649 19087-19087/org.tensorflow.demo W/tensorflow.dem: JIT profile information will not be recorded: profile file does not exits.
2019-09-15 11:38:40.650 19087-19087/org.tensorflow.demo W/tensorflow.dem: JIT profile information will not be recorded: profile file does not exits.
2019-09-15 11:38:40.655 19087-19087/org.tensorflow.demo I/chatty: uid=10102(org.tensorflow.demo) identical 5 lines
2019-09-15 11:38:40.655 19087-19087/org.tensorflow.demo W/tensorflow.dem: JIT profile information will not be recorded: profile file does not exits.
2019-09-15 11:38:40.737 19087-19087/org.tensorflow.demo I/InstantRun: starting instant run server: is main process
2019-09-15 11:38:40.891 19087-19087/org.tensorflow.demo D/tensorflow: CameraActivity: onCreate org.tensorflow.demo.ClassifierActivity@ddf260d
2019-09-15 11:38:41.038 19087-19087/org.tensorflow.demo D/tensorflow: CameraActivity: onStart org.tensorflow.demo.ClassifierActivity@ddf260d
2019-09-15 11:38:41.041 19087-19087/org.tensorflow.demo D/tensorflow: CameraActivity: onResume org.tensorflow.demo.ClassifierActivity@ddf260d
2019-09-15 11:38:41.058 19087-19087/org.tensorflow.demo D/OpenGLRenderer: HWUI GL Pipeline
2019-09-15 11:38:41.111 19087-19087/org.tensorflow.demo D/tensorflow: CameraActivity: onPause org.tensorflow.demo.ClassifierActivity@ddf260d
2019-09-15 11:38:41.112 19087-19087/org.tensorflow.demo D/tensorflow: CameraActivity: Requesting finish
2019-09-15 11:38:41.349 19087-19106/org.tensorflow.demo I/ConfigStore: android::hardware::configstore::V1_0::ISurfaceFlingerConfigs::hasWideColorDisplay retrieved: 0
2019-09-15 11:38:41.354 19087-19106/org.tensorflow.demo I/ConfigStore: android::hardware::configstore::V1_0::ISurfaceFlingerConfigs::hasHDRDisplay retrieved: 0
2019-09-15 11:38:41.354 19087-19106/org.tensorflow.demo I/OpenGLRenderer: Initialized EGL, version 1.4
2019-09-15 11:38:41.355 19087-19106/org.tensorflow.demo D/OpenGLRenderer: Swap behavior 1
2019-09-15 11:38:41.355 19087-19106/org.tensorflow.demo W/OpenGLRenderer: Failed to choose config with EGL_SWAP_BEHAVIOR_PRESERVED, retrying without...
2019-09-15 11:38:41.355 19087-19106/org.tensorflow.demo D/OpenGLRenderer: Swap behavior 0
2019-09-15 11:38:41.437 19087-19106/org.tensorflow.demo D/EGL_emulation: eglCreateContext: 0xf1143360: maj 2 min 0 rcv 2
2019-09-15 11:38:41.470 19087-19106/org.tensorflow.demo D/EGL_emulation: eglMakeCurrent: 0xf1143360: ver 2 0 (tinfo 0xf10b1aa0)
2019-09-15 11:38:41.579 19087-19106/org.tensorflow.demo D/EGL_emulation: eglMakeCurrent: 0xf1143360: ver 2 0 (tinfo 0xf10b1aa0)
2019-09-15 11:38:42.612 19087-19106/org.tensorflow.demo D/EGL_emulation: eglMakeCurrent: 0xf1143360: ver 2 0 (tinfo 0xf10b1aa0)
2019-09-15 11:38:42.769 19087-19087/org.tensorflow.demo D/tensorflow: CameraActivity: onStop org.tensorflow.demo.ClassifierActivity@ddf260d
2019-09-15 11:38:42.784 19087-19087/org.tensorflow.demo D/tensorflow: CameraActivity: onDestroy org.tensorflow.demo.ClassifierActivity@ddf260d
2019-09-15 11:39:02.133 19087-19087/org.tensorflow.demo W/ActivityThread: handleWindowVisibility: no activity for token android.os.BinderProxy@9aa9a4b
2019-09-15 11:39:02.493 19087-19087/org.tensorflow.demo D/tensorflow: CameraActivity: onCreate org.tensorflow.demo.DetectorActivity@c9152e6
2019-09-15 11:39:02.631 19087-19087/org.tensorflow.demo I/CameraManagerGlobal: Connecting to camera service
2019-09-15 11:39:02.698 19087-19087/org.tensorflow.demo I/tensorflow: CameraActivity: Camera API lv2?: false
2019-09-15 11:39:02.864 19087-19087/org.tensorflow.demo W/tensorflow.dem: Verification of java.lang.Object org.tensorflow.demo.AutoFitTextureView.access$super(org.tensorflow.demo.AutoFitTextureView, java.lang.String, java.lang.Object[]) took 128.972ms
2019-09-15 11:39:02.911 19087-19087/org.tensorflow.demo D/tensorflow: CameraActivity: onStart org.tensorflow.demo.DetectorActivity@c9152e6
2019-09-15 11:39:02.912 19087-19087/org.tensorflow.demo D/tensorflow: CameraActivity: onResume org.tensorflow.demo.DetectorActivity@c9152e6
2019-09-15 11:39:03.413 19087-19087/org.tensorflow.demo I/tensorflow: CameraConnectionFragment: Desired size: 640x480, min size: 480x480
2019-09-15 11:39:03.414 19087-19087/org.tensorflow.demo I/tensorflow: CameraConnectionFragment: Valid preview sizes: [640x480, 1280x720, 1280x960]
2019-09-15 11:39:03.414 19087-19087/org.tensorflow.demo I/tensorflow: CameraConnectionFragment: Rejected preview sizes: [352x288, 320x240, 176x144]
2019-09-15 11:39:03.414 19087-19087/org.tensorflow.demo I/tensorflow: CameraConnectionFragment: Exact size match found.
2019-09-15 11:39:03.470 19087-19087/org.tensorflow.demo W/tensorflow: ImageUtils: Native library not found, native RGB -> YUV conversion may be unavailable.
2019-09-15 11:39:03.512 19087-19106/org.tensorflow.demo D/EGL_emulation: eglMakeCurrent: 0xf1143360: ver 2 0 (tinfo 0xf10b1aa0)
2019-09-15 11:39:04.495 19087-19087/org.tensorflow.demo W/tensorflow: TensorFlowObjectDetectionAPIModel: ???
2019-09-15 11:39:04.495 19087-19087/org.tensorflow.demo W/tensorflow: TensorFlowObjectDetectionAPIModel: Awake
2019-09-15 11:39:04.495 19087-19087/org.tensorflow.demo W/tensorflow: TensorFlowObjectDetectionAPIModel: 
2019-09-15 11:39:04.496 19087-19087/org.tensorflow.demo W/tensorflow: TensorFlowObjectDetectionAPIModel: Fatigue
2019-09-15 11:39:04.528 19087-19087/org.tensorflow.demo I/TensorFlowInferenceInterface: Checking to see if TensorFlow native methods are already loaded
2019-09-15 11:39:04.529 19087-19087/org.tensorflow.demo E/tensorflow.dem: No implementation found for long org.tensorflow.contrib.android.RunStats.allocate() (tried Java_org_tensorflow_contrib_android_RunStats_allocate and Java_org_tensorflow_contrib_android_RunStats_allocate__)
2019-09-15 11:39:04.529 19087-19087/org.tensorflow.demo I/TensorFlowInferenceInterface: TensorFlow native methods not found, attempting to load via tensorflow_inference
2019-09-15 11:39:04.665 19087-19098/org.tensorflow.demo I/tensorflow.dem: Background concurrent copying GC freed 1106(85KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 2MB/5MB, paused 1.666ms total 184.225ms
2019-09-15 11:39:04.747 19087-19087/org.tensorflow.demo W/native: cpu_feature_guard.cc:35 The TensorFlow library was compiled to use SSE instructions, but these aren't available on your machine.
2019-09-15 11:39:04.747 19087-19087/org.tensorflow.demo W/native: cpu_feature_guard.cc:35 The TensorFlow library was compiled to use SSE2 instructions, but these aren't available on your machine.
2019-09-15 11:39:04.747 19087-19087/org.tensorflow.demo W/native: cpu_feature_guard.cc:35 The TensorFlow library was compiled to use SSE3 instructions, but these aren't available on your machine.
2019-09-15 11:39:04.771 19087-19087/org.tensorflow.demo I/TensorFlowInferenceInterface: Successfully loaded TensorFlow native methods (RunStats error may be ignored)
2019-09-15 11:39:05.861 19087-19087/org.tensorflow.demo E/tensorflow: CameraActivity: Exception!
    java.lang.RuntimeException: Failed to load model from 'file:///android_asset/ssd.pb'
        at org.tensorflow.contrib.android.TensorFlowInferenceInterface.<init>(TensorFlowInferenceInterface.java:113)
        at org.tensorflow.demo.TensorFlowObjectDetectionAPIModel.create(TensorFlowObjectDetectionAPIModel.java:91)
        at org.tensorflow.demo.DetectorActivity.onPreviewSizeChosen(DetectorActivity.java:157)
        at org.tensorflow.demo.CameraActivity.onPreviewFrame(CameraActivity.java:120)
        at android.hardware.Camera$EventHandler.handleMessage(Camera.java:1209)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6669)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: java.io.IOException: Not a valid TensorFlow Graph serialization: NodeDef mentions attr 'half_pixel_centers' not in Op<name=ResizeBilinear; signature=images:T, size:int32 -> resized_images:float; attr=T:type,allowed=[DT_INT8, DT_UINT8, DT_INT16, DT_UINT16, DT_INT32, DT_INT64, DT_BFLOAT16, DT_HALF, DT_FLOAT, DT_DOUBLE]; attr=align_corners:bool,default=false>; NodeDef: {{node Preprocessor/map/while/ResizeImage/resize/ResizeBilinear}}. (Check whether your GraphDef-interpreting binary is up to date with your GraphDef-generating binary.).
        at org.tensorflow.contrib.android.TensorFlowInferenceInterface.loadGraph(TensorFlowInferenceInterface.java:561)
        at org.tensorflow.contrib.android.TensorFlowInferenceInterface.<init>(TensorFlowInferenceInterface.java:105)
        at org.tensorflow.demo.TensorFlowObjectDetectionAPIModel.create(TensorFlowObjectDetectionAPIModel.java:91) 
        at org.tensorflow.demo.DetectorActivity.onPreviewSizeChosen(DetectorActivity.java:157) 
        at org.tensorflow.demo.CameraActivity.onPreviewFrame(CameraActivity.java:120) 
        at android.hardware.Camera$EventHandler.handleMessage(Camera.java:1209) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.app.ActivityThread.main(ActivityThread.java:6669) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)

**Comment by dm961027** [comment]
I have already solved this problem , I use tensorflow 1.13.1 replaced tensorflow 1.14.0 and add 
 implementation 'org.tensorflow:tensorflow-android:1.13.1'  thank you everyone

**Comment by jvishnuvardhan** [comment]
Thanks @dm961027 I am closing the issue as it was resolved. Please feel free to open it if the issue persists again. Thanks!

**Comment by fcu-d0571783** [comment]
@dm961027  
implementation 'org.tensorflow:tensorflow-android:1.13.1' 现在最高版本就1.13.1
所以训练模型，转成tflite都得用tensorflow 1.13.1的版本吗

**Comment by SirPhemmiey** [comment]
@dm961027 @fcu-d0571783 Did you had to retrain the model again with tensorflow 1.13.1 for it to work or you just downgraded your tensorflow version and it worked?
